include/arch/mips/danube/linker.ld

```diff
@@ -0,0 +1,383 @@
+/*
+ * Copyright Imagination Technologies Limited and/or its
+ *                      affiliated group companies.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ * this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * 3. Neither the name of the copyright holder nor the names of its
+ * contributors may be used to endorse or promote products derived from this
+ * software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+*/
+/*
+ * A platform and target independent link script to produce UHI
+ * compliant binaries with varying levels of system initialization
+ * support.
+ */
+
+ #define _LINKER
+ #define _ASMLANGUAGE
+
+ #include <autoconf.h>
+ #include <linker/sections.h>
+ #include <generated_dts_board.h>
+
+ #include <linker/linker-defs.h>
+ #include <linker/linker-tool.h>
+
+ #ifdef CONFIG_XIP
+ #define ROMABLE_REGION              ROM
+ #else
+ #define ROMABLE_REGION              RAM
+ #endif
+ #define RAMABLE_REGION              RAM
+
+ MEMORY
+ {
+     ROM (rx)  : ORIGIN = CONFIG_FLASH_BASE_ADDRESS, LENGTH = \
+         CONFIG_FLASH_SIZE * 1024
+     RAM (rwx) : ORIGIN = CONFIG_SRAM_BASE_ADDRESS, LENGTH = \
+        CONFIG_SRAM_SIZE * 1024
+     /* Used by and documented in include/linker/intlist.ld */
+     IDT_LIST  (wx)      : ORIGIN = 0xFFFFF7FF, LENGTH = 2K
+ }
+
+OUTPUT_FORMAT("elf32-tradlittlemips", "elf32-tradbigmips", \
+    "elf32-tradlittlemips")
+GROUP(-lc -luhi -lgcc)
```

Comment:
I am trying with a toolchain built with xtools, everything builds fine but at the final linking stage I hit this, first the output format is unknown (changing to elf32-littlemips works) and then I can't link with -luhi. What is libuhi and and do we need this with the hal in ext/?